decision-records/DR-0001-why-docusaurus.md

```diff
@@ -0,0 +1,105 @@
+| :---------- | :-------------------------------------------------------------------------------------- |
+| **PR #**    | [6](https://github.com/blindnet-io/blindnet-docs/issues/6)                              |
+| **Author** & **Sponsor** | Marko Stakic (marko@blindnet.io)                                           |
+| **Lead Reviewer** | Noël Macé (noel@blindnet.io)                                                      |
+
+## Context and Problem Statement
+
+As specified in [this epic](https://github.com/blindnet-io/communication-management/issues/14), documentaion pages created in [MkDocs](https://www.mkdocs.org) [Material](https://squidfunk.github.io/mkdocs-material/) theme look outdated and its customization is cumbersome.
+
+Which OSS documentation rendering engine offers the [needed feature set](https://github.com/blindnet-io/communication-management/issues/63#issuecomment-1080364433) as well as a better look and feel without requiring important changes in the doc itself?
+
+## Decision Drivers
+
+Since the devkit has implementations in multiple programming languages, having content tabs component (tab per language) is crucial.
+
+Other drivers:
+- what features are available
+- ease of use
+- customization capabilities
+- look and UX
+
+## Considered Options
+
+- [Docusaurus](https://docusaurus.io)
+- [Rocket](https://rocket.modern-web.dev)
+- [docsify](https://docsify.js.org)
+- [mdBook](https://rust-lang.github.io/mdBook)
+- [GitBook](https://www.gitbook.com)
+- [Read the Docs](https://readthedocs.org)
+
+## Decision Outcome
+
+Docusaurus is for us the best suited project for rendering documentation. It is superior in terms of easiness of use, feature richness, overall look and has a large community of users and maintainers.
+
+## Pros and Cons of the Options
+
+### Docusaurus
+
+#### pros
+- docs can be written in both Markdown and [React](https://reactjs.org)
+- large community
+- feature rich
+- beautiful theme and modern look
+- integration with [Algolia](https://www.algolia.com) for searching the docs
+
+#### cons
+- only one theme
+- not so great plugins for offline search
+
+### Rocket
+
+#### cons
+- their website didn't work well, some blocks were not rendering in Firefox
+- maintenance issues
+- general static site builder, not specialised for documentation
+
+### docsify
+
+#### pros
+- easily customizable with Markdown and [Vue](https://vuejs.org)
+
+#### cons
+- no native content tabs (plugin for content tabs looks awkward)
+- less features than other options
+- simplistic result
+
+### mdBook
+
+#### pros
+- good local search engine
+- native executable code blocks for [Rust](https://www.rust-lang.org)
+
+#### cons
+- made specifically for Rust
+- no content tabs
+
+### GitBook
+
+#### pros
+- easy to set-up
+- integration with third party services (Slack, Github, etc.)
+- feature rich
+- beautiful, modern look
+
+#### cons
+- premium product
```

Comment:
this may require a bit more information :wink:

Reply:
Updated in c15ef210849ea037b8c269c43425bd5046a9e72d